de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/IntellijProjectLifecycle.java

```diff
@@ -24,30 +26,40 @@
 public class IntellijProjectLifecycle extends AbstractContextLifecycle {
 
   private static IntellijProjectLifecycle instance;
+  private static ProjectWrapper projectWrapper;
 
   /**
-   * Creates a new IntellijProjectLifecycle singleton instance from a project.
+   * Returns the current instance of IntellijProjectLifecycle.
    *
-   * @param project
-   * @return
+   * <p>IntellijProjectLifecycle is singleton, meaning this call will always return the same object.
+   * If a new project object is passed, it will replace the currently held project object in the
+   * plugin context.
+   *
+   * @param project the project object to put into the plugin context
+   * @return the singleton instance of the IntellijProjectLifecycle
    */
-  public static synchronized IntellijProjectLifecycle getInstance(Project project) {
-    instance = new IntellijProjectLifecycle(project);
+  public static synchronized IntellijProjectLifecycle getInstance(@NotNull Project project) {
```

Comment:
Add information in the commit message that you make this class a real singleton

Reply:
Done.